parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftRecordConverter.java

```diff
@@ -445,6 +600,26 @@ public FieldEnumConverter(List<TProtocol> events, ThriftField field) {
       }
     }
 
+     @Override
+     protected ParquetProtocol decodeFromDictionary(Dictionary d, int i) {
+       Binary value = d.decodeToBinary(i);
+       final Integer id = enumLookup.get(value);
```

Comment:
I think some of these could be refactored to reduce duplication. What about a `event` method that creates the ParquetProtocol object? That would reduce the size of this code quite a bit.